cpp/src/io/utilities/parsing_utils.cuh

```diff
@@ -501,33 +506,225 @@ __inline__ __device__ std::pair<char const*, char const*> trim_whitespaces_quote
+}
+
+template <typename T, std::enable_if_t<cudf::is_timestamp<T>()>* = nullptr>
+__inline__ __device__ T decode_value(char const* begin,
+                                     char const* end,
+                                     parse_options_view const& opts)
+{
+  return to_timestamp<T>(begin, end, opts.dayfirst);
+}
+
+template <typename T, std::enable_if_t<cudf::is_duration<T>()>* = nullptr>
+__inline__ __device__ T decode_value(char const* begin, char const* end, parse_options_view const&)
+{
+  return to_duration<T>(begin, end);
+}
+
+// The purpose of these is merely to allow compilation ONLY
+template <>
+__inline__ __device__ cudf::string_view decode_value(const char*,
+                                                     const char*,
+                                                     parse_options_view const&)
+{
+  return cudf::string_view{};
+}
+
+template <>
+__inline__ __device__ cudf::dictionary32 decode_value(const char*,
+                                                      const char*,
+                                                      parse_options_view const&)
+{
+  return cudf::dictionary32{};
+}
+
+template <>
+__inline__ __device__ cudf::list_view decode_value(const char*,
+                                                   const char*,
+                                                   parse_options_view const&)
+{
+  return cudf::list_view{};
+}
+template <>
+__inline__ __device__ cudf::struct_view decode_value(const char*,
+                                                     const char*,
+                                                     parse_options_view const&)
+{
+  return cudf::struct_view{};
+}
+
+template <>
+__inline__ __device__ numeric::decimal32 decode_value(const char*,
+                                                      const char*,
+                                                      parse_options_view const&)
 {
-  return thrust::find_if(thrust::seq, begin, end, [](auto elem) { return elem != ' '; });
+  return numeric::decimal32{};
 }
 
+template <>
+__inline__ __device__ numeric::decimal64 decode_value(const char*,
+                                                      const char*,
+                                                      parse_options_view const&)
+{
+  return numeric::decimal64{};
+}
+
+template <>
+__inline__ __device__ numeric::decimal128 decode_value(const char*,
+                                                       const char*,
+                                                       parse_options_view const&)
+{
+  return numeric::decimal128{};
+}
+
+struct ConvertFunctor {
+  /**
+   * @brief Dispatch for numeric types whose values can be convertible to
+   * 0 or 1 to represent boolean false/true, based upon checking against a
+   * true/false values list.
+   *
+   * @return bool Whether the parsed value is valid.
+   */
+  template <typename T,
+            std::enable_if_t<std::is_integral_v<T> and !std::is_same_v<T, bool> and
```

Comment:
It's nice to use [`CUDF_ENABLE_IF`](https://github.com/rapidsai/cudf/blob/1d488098366eff2d885bce85f6de1c5e574aae7e/cpp/include/cudf/utilities/traits.hpp#L49) for SFINAE if possible. Applies in a few places.

Reply:
Agree, looks much nicer. Fixed 👍